component/src/main/java/io/siddhi/extension/io/file/function/FileArchiveExtension.java

```diff
@@ -0,0 +1,225 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.siddhi.extension.io.file.function;
+
+import io.siddhi.annotation.Example;
+import io.siddhi.annotation.Extension;
+import io.siddhi.annotation.Parameter;
+import io.siddhi.annotation.ReturnAttribute;
+import io.siddhi.annotation.util.DataType;
+import io.siddhi.core.config.SiddhiQueryContext;
+import io.siddhi.core.exception.SiddhiAppRuntimeException;
+import io.siddhi.core.executor.ExpressionExecutor;
+import io.siddhi.core.executor.function.FunctionExecutor;
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.snapshot.state.StateFactory;
+import io.siddhi.query.api.definition.Attribute;
+import io.siddhi.query.api.exception.SiddhiAppValidationException;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static io.siddhi.query.api.definition.Attribute.Type.BOOL;
+import static io.siddhi.query.api.definition.Attribute.Type.STRING;
+
+/**
+ * This extension can be used to archive files.
+ */
+@Extension(
+        name = "archive",
+        namespace = "file",
+        description = "This function archives files at a given file path (recursively if given).\n",
+        parameters = {
+                @Parameter(
+                        name = "source.path",
+                        description = "The source path of the folder or the file to be archived.",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "destination.folder.path",
+                        description = "The destination path of the archived file.",
+                        type = DataType.STRING
+                ),
+                @Parameter(
+                        name = "archive.recursively",
+                        description = "This parameter will allow recursive folder archive.",
+                        type = DataType.BOOL
+                )
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        description = "The success of the file archive.",
+                        type = DataType.BOOL
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "from ArchiveFileStream\n" +
+                                "select file:archive('/User/wso2/to_be_archived', " +
+                                    "'User/wso2/archive_destination/file.zip', ' + true + ') as isArchived \n" +
```

Comment:
'User/wso2/archive_destination/file.zip', not an absolute path? so where it will be retrieved

Reply:
@suhothayan It should be an absolute path. This is fixed in https://github.com/siddhi-io/siddhi-io-file/pull/72/commits/4eaba8254220943cd983761f19554006840f5d17